app/controllers/applications/process_controller.rb

```diff
@@ -148,13 +134,31 @@ def form_params(type)
     end
 
     def application
-      @appication ||= Application.find(params[:application_id])
+      @application ||= Application.find(params[:application_id])
     end
 
     def user_jurisdictions
       current_user.office.jurisdictions
     end
 
+    def benefit_check_runner
+      @benefit_check_runner ||= BenefitCheckRunner.new(application)
+    end
+
+    def benefit_check_and_redirect(benefits)
+      if benefits
+        benefit_check_runner.run
+
+        if benefit_check_runner.can_override?
+          redirect_to application_benefit_override_paper_evidence_path(application)
+        else
+          redirect_to(action: :summary)
+        end
+      else
+        redirect_to(action: :income)
+      end
+    end
+
     def calculate_income
```

Comment:
I know that you haven't touched this method. But it could be 'shaved off' by moving the "whole content of the method" back to where `calculate_income` is called.